influxql/call_iterator.go

```diff
@@ -146,26 +170,34 @@ func integerMaxReduce(prev, curr *IntegerPoint, opt *reduceOptions) (int64, int6
 func newSumIterator(input Iterator, opt IteratorOptions) (Iterator, error) {
 	switch input := input.(type) {
 	case FloatIterator:
-		return &floatReduceIterator{input: newBufFloatIterator(input), opt: opt, fn: floatSumReduce}, nil
+		createFn := func() (FloatPointAggregator, FloatPointReducer) {
+			fn := NewFloatFuncReducer(floatSumReduce)
+			return fn, fn
+		}
+		return &floatReduceIterator{input: newBufFloatIterator(input), opt: opt, create: createFn}, nil
 	case IntegerIterator:
-		return &integerReduceIterator{input: newBufIntegerIterator(input), opt: opt, fn: integerSumReduce}, nil
+		createFn := func() (IntegerPointAggregator, IntegerPointReducer) {
+			fn := NewIntegerFuncReducer(integerSumReduce)
+			return fn, fn
+		}
+		return &integerReduceIterator{input: newBufIntegerIterator(input), opt: opt, create: createFn}, nil
 	default:
 		return nil, fmt.Errorf("unsupported sum iterator type: %T", input)
 	}
 }
 
 // floatSumReduce returns the sum prev value & curr value.
-func floatSumReduce(prev, curr *FloatPoint, opt *reduceOptions) (int64, float64, []interface{}) {
+func floatSumReduce(prev, curr *FloatPoint) (int64, float64, []interface{}) {
 	if prev == nil {
-		return curr.Time, curr.Value, nil
+		return ZeroTime, curr.Value, nil
 	}
 	return prev.Time, prev.Value + curr.Value, nil
 }
 
 // integerSumReduce returns the sum prev value & curr value.
-func integerSumReduce(prev, curr *IntegerPoint, opt *reduceOptions) (int64, int64, []interface{}) {
+func integerSumReduce(prev, curr *IntegerPoint) (int64, int64, []interface{}) {
 	if prev == nil {
-		return curr.Time, curr.Value, nil
+		return ZeroTime, curr.Value, nil
```

Comment:
Sum's don't have a time. The time returned is only for a specific point. I'm not sure why this ever returned a time since it was ignored and wrong to begin with.

I decided that returning `ZeroTime` for functions where it made no sense was the best compromise rather than building an entirely new iterator for it.